File: Origin/reconstructed-project/src/input/BattleInputCommand.js
'use strict';
const BattleInputCommandType = Object.freeze({ PURCHASE_AND_PLACE:'PurchaseAndPlace', BEGIN_DRAG:'BeginDrag', MOVE_DRAG:'MoveDrag', COMMIT_PLACEMENT:'CommitPlacement', CANCEL_DRAG:'CancelDrag', MOVE_UNIT:'MoveUnit', MERGE_UNITS:'MergeUnits', REFRESH:'Refresh' });
class BattleInputCommand { constructor(type,payload={}){if(!Object.values(BattleInputCommandType).includes(type))throw new Error(`Unknown BattleInputCommand ${type}`);this.type=type;this.payload={...payload};} }
module.exports={BattleInputCommandType,BattleInputCommand};

File: Origin/reconstructed-project/unity-handoff/csharp-reference/BattleInputCommand.cs
namespace Game.Combat.Input
{
    public enum BattleInputCommandType
    {
        PurchaseAndPlace,
        BeginDrag,
        MoveDrag,
        CommitPlacement,
        CancelDrag,
        MoveUnit,
        MergeUnits,
        Refresh
    }

    public readonly record struct GridPosition(int X, int Y);
    public sealed record BattleInputCommand(BattleInputCommandType Type, object Payload);
    public sealed record PurchaseAndPlacePayload(bool PlayerSide, int Slot, GridPosition Position);
    public sealed record MergeUnitsPayload(int SourceId, int TargetId);
    public sealed record MoveUnitPayload(int UnitId, GridPosition Position);
}
